students/guillaume/session03/mailroom.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+def read_data(textfile):
+    '''
+    Read data from a textfile about the donor and the
+    amount of donations
+    Pattern is as follow:
+    '''
+    file = open(textfile, 'r')
+    donors = list()
+    for i in list(file.readlines()):
+        donors.append((i.rstrip()).split(','))
+        for i, s in enumerate(donors[-1]):
+            donors[-1][i] = s.lstrip()
+            if donors[-1][i].isdecimal():
+                donors[-1][i] = int(s)
+    file.close()
+    return donors
+
+
+def write_data(textfile, donors):
+    file = open(textfile, 'w')
+    for donor in donors:
+        donor_str = ', '.join(map(str, donor)) + '\n'
+        file.write(donor_str)
+    file.close()
+
+
+def add_donation(ind, donors):
+    '''
+    '''
+    boolean = True
+    while boolean:
+        donation = input('\nPlease add a donation amount?\n')
+        if donation.isdecimal():
+            # need to work on test for float donation
+            boolean = False
+            donation = int(donation)
+            donors[ind].append(donation)
+    return donors, donation
+
+
+def email_thank_you(donor, donation, charity='local charity'):
+    '''
+    '''
+    email = '''
+            Dear {},
+
+            I would like to thank you personnaly for the generous
+            donation of ${} that you have made recently.
+            Your committment to our cause is very valuable to us.
+
+            Best regards,
+
+                                    John Adams
+                                    CEE of charity {}
+            '''.format(donor, donation, charity)
+    return email
+
+
+def send_thank_you(textfile):
+    '''
+    '''
+    donors = read_data(textfile)
+    names = ['{} {}'.format(i[0], i[1]) for i in donors]
+    boolean = True
+    # Menu for Name cases
+    while boolean:
+        name = input('Plese enter a full name?\n')
+        if name == 'list':
+            print('\n')
+            for i in names:
+                print(i)
+        elif name in names:
+            boolean = False
+            donors, donation = add_donation(names.index(name), donors)
+        elif ' ' in name:
+            # Checking for Last & First Name - very primitive
+            boolean = False
+            names.append(name)
+            donors.append(name.split(' '))
+            donors, donation = add_donation(-1, donors)
+        else:
+            print('Make sure you enter Last & first name\n')
+
+    write_data(textfile, donors)
+    email = email_thank_you(name, donation)
+    return email
+
+
+def report(textfile):
+    donors = read_data(textfile)
```

Comment:
probably better to load the data into a data structure in memory, and then work with it in all the functions.